File: solutions/python/blind-75/array/contains_duplicate.py
class Solution:
    def containsDuplicate(self, nums):
        """
        Solution overview:
        Track numbers we've seen so far. If we hit an element that we've already seen, return
        true.
        """
        seen = set()

        for num in nums:
            if num in seen:
                return True

            seen.add(num)

        return False

File: solutions/javascript/general-algos/arrays/containsDuplicate.js
/*
Solution overview:
Track numbers we've seen so far. If we hit an element that we've already seen, return
true.
*/
const containsDuplicate = (nums) => {
  const seen = new Set();
  
  for (const num of nums) {
      if (seen.has(num)) return true;
      
      seen.add(num);
  }
  
  return false;
};